### PR TITLE
Feature/ab2d 4266 job contract bus key

### PR DESCRIPTION
**Pull request by nflearl**
**JIRA Tickets:**

[AB2D-4266](https://jira.cms.gov/browse/AB2D-4266) - Convert Job reference to Contract to Business Key

***Related Tickets***

[AB2D-4109](https://jira.cms.gov/browse/AB2D-4109) - Adding a not null constraint on the contract_number field in job moves us along to this goal.
 
### What Does This PR Do?
Removes the foreign key from job to contract and replaces it with a business key.  If one does not already possess a contract reference and needs information about the contract, then the contract will need to be fetched.

### What Should Reviewers Watch For?
Unnecessary fetches of contracts where we might already have it in memory.

### Database Changes
Added and populated business key column
Dropped foreign key column

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [x] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [x] Code checked for PHI/PII exposure

### CODE REVIEW

File: worker/src/main/java/gov/cms/ab2d/worker/processor/JobPreProcessorImpl.java [hunk 1]
```diff
@@ -57,26 +61,28 @@ public Job preprocess(String jobUuid) {
             throw new IllegalArgumentException(errMsg);
         }
 
-        Optional sinceValue = Optional.ofNullable(job.getSince());
+        Optional<Contract> contractOptional = contractRepository.findContractByContractNumber(job.getContractNumber());
```
Comment (sb-wnyffenegger) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/JobPreProcessorImpl.java:64]: Shouldn't we throw an error if the contract can't be found? Seems like a pretty big issue if the contract number for a job doesn't match an existing contract. In fact, we won't be able to run anything if that is the case. I would just thrown an exception here and fail the job outright.

**Review by sb-wnyffenegger — COMMENTED** [review]
Mostly minor comments.

Larger questions: for the coverage tests is this going to conflict heavily with Dugan's changes creating the coverage module?

File: worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java [hunk 2]
```diff
@@ -29,15 +30,14 @@
...
-import static gov.cms.ab2d.common.model.JobStatus.CANCELLED;
 import static gov.cms.ab2d.common.util.Constants.CONTRACT_LOG;
 import static gov.cms.ab2d.common.util.EventUtils.getOrganization;
 import static gov.cms.ab2d.fhir.BundleUtils.EOB;
 import static net.logstash.logback.argument.StructuredArguments.keyValue;
 
 @Slf4j
 @Service
-@SuppressWarnings({"PMD.TooManyStaticImports", "java:S2142"}) //java:S2142: "InterruptedException" should not be ignored
+@SuppressWarnings({"PMD.TooManyStaticImports", "java:S2142", "squid:S00107"}) //java:S2142: "InterruptedException" should not be ignored
```
[the remark this answers is not in the document]
Comment (nflearl) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java:40]: On my final cleanup pass, determined that this has no effect and have removed it.

File: worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java [hunk 3]
```diff
@@ -410,9 +410,9 @@ public void loadMappingJob() {
      */
     @Trace(metricName = "EnrollmentIsAvailable", dispatcher = true)
     @Override
-    public boolean isCoverageAvailable(Job job) throws InterruptedException {
+    public boolean isCoverageAvailable(Job job, Contract contract) throws InterruptedException {
 
-        String contractNumber = job.getContract().getContractNumber();
+        String contractNumber = job.getContractNumber();
```
[the remark this answers is not in the document]
Comment (nflearl) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java:415]: Sure.  Done

**Comment by sb-dkniesteadt** [comment]
This will conflict with the coverage code. I have agreed to deal with the conflict.

**Review by sb-dkniesteadt — APPROVED** [review]

**Review by sharshar-sb — APPROVED** [review]

**Pull request merged**